### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by nnrogers515**

### CODE REVIEW

**Comment by nnrogers515** [comment]
Pipeline is currently bare minimal, have to add some testing stuff for it to really be useful with the working app connections. Currently it just renders the pages lol.

**Comment by nnrogers515** [comment]
Also ignore the .expressOptionalblahblah, that's for later if we want to use express instead but it is entirely unnecessary

**Review by GabriellaXue — CHANGES_REQUESTED** [review]
Seems like Express is doing the same thing as React, and currently, they don't have any intersection with each other. Can we take out the express folder for now if we are not using it? It's hard to peer review 70 files at the same time.  

Also, may I ask why are we considering Express?

**Comment by nnrogers515** [comment]
Express is supposed to have a lot of nice backend functionality and an ease of concurrent execution for frontend and backend (if you look at the package.json scripts I had to utilize nohup for starting the server and another command for starting the frontend, express doesn't require that as it can do both from my understanding). I will remove the files from this merge request though.

**Comment by nnrogers515** [comment]
Okay files have been updated! Should be good to go, let me know if you have any more comments!

**Comment by GabriellaXue** [comment]
Oops, can’t believe I just saw it! Sorry about that! 

I have one more question tho, I don’t know if we want to keep backend separate from frontend. I used the server and client folders preciously to separate parts of the stack. Right now our server is fairly simple but feels like, when we add more functionalities to it, having backend and frontend separate would lessen the chance of breaking the entire website and would be a more intuitive way for debugging. Let me know what do u guys think. @nnrogers515 @jamesbayus

**Comment by GabriellaXue** [comment]
To install the dependencies, do I just need yarn install?

**Comment by nnrogers515** [comment]
Yes to `yarn install` I believe `npm install` would work splendid as well. As for the other question, let me rephrase. The backend and frontend code isn't separate so to speak but the execution is separate commands. From my understanding, there isn't a process that would be able to handle both the server and the frontend react with the way it is currently setup, this would be why we would switch to express since it has these functionalities. However they are likely doing the same thing behind the scenes, and I don't know how much impact it would have on debugging.

In reality, one process will just maintain a server for the websocket frontend to connect to, I believe NGINX is utilized to streamline frontend/backend communication in some cases which we can explore later. There is also a likelihood I missed something so if you have an implementation that you think would be better feel free to let me know and we can go through those benefits :).

**Review by GabriellaXue — APPROVED** [review]
